src/metadataExt.ts

```diff
@@ -186,6 +187,21 @@ export class MetadataModuleExt {
         });
     }
 
+    private getDashboardDetailObject(
+        updatedContent: IAnalyticalDashboardContent,
+        filterContext: string,
+    ): IAnalyticalDashboardContent {
+        if (isEmpty(updatedContent.layout)) {
```

Comment:
Stylistic detail: there is a large overlap between the two branches: the only difference is that if there is no layout, you don't want to copy it, right? You could make this less verbose by doing something like

```ts
const { layout } = updatedContent;
return {
  ...updatedContent,
  filterContext,
  widgets: [...updatedContent.widgets],
  ...(isEmpty(layout) ? {} : { layout }),
}
```

i.e. effectively making the `if`only on the one property you don't want to include sometimes. Your version might be more readable, so change it only if this seems better to you ;) I just wanted to show you this pattern, because we use it sometimes and this seemed as a good place to show this :)